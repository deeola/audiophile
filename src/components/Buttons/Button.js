import React from "react";
import { Buttons } from "./ButtonStyling";

export const Button = ({ label, link, datatestid }) => {
  return <Buttons data-testid={datatestid}  color= 'primary' hover='primary' to={link}>{label}</Buttons>;
};


export const SecondaryButton = ({ label, link, datatestid }) => {
    return <Buttons data-testid={datatestid} color='secondary' hover='secondary'  to={link}>{label}</Buttons>;
};

export const TetiaryButton = ({ label, link, span, datatestid }) => {
    return <Buttons data-testid={datatestid} color= 'tertiary'  hover='tertiary' to={link}>{label}<span>{span}</span></Buttons>;
};


export const BlackButton = ({ label, link, datatestid }) => {
    return <Buttons data-testid={datatestid}  to={link}>{label}</Buttons>;
};




