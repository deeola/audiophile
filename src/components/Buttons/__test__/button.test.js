import React from "react";
import { render, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { Button, SecondaryButton, TetiaryButton } from "../Button";
import { BrowserRouter } from "react-router-dom";
import renderer from "react-test-renderer";

afterEach(cleanup);

it("renders primary button with correct text", () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <Button label={"SEE PRODUCT"} datatestid='primarybutton' />
    </BrowserRouter>
  );
  expect(getByTestId("primarybutton")).toHaveTextContent("SEE PRODUCT");
});

it("renders secondary button with correct text", () => {
  const { getByTestId } = render(
    <BrowserRouter>
      <SecondaryButton datatestid={'secondarybutton'} label={"SEE PRODUCT"} />
    </BrowserRouter>
  );
  expect(getByTestId("secondarybutton")).toHaveTextContent("SEE PRODUCT");
});

it("renders tetiary button with correct text", () => {
    const { getByTestId } = render(
      <BrowserRouter>
        <TetiaryButton datatestid={'tetiarybutton'} label={"SEE PRODUCT"} />
      </BrowserRouter>
    );
    expect(getByTestId("tetiarybutton")).toHaveTextContent("SEE PRODUCT");
    expect(getByTestId("tetiarybutton")).toContainHTML('span');
  });

// Snapshots

it("Primary Button matches snapshot", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <Button label={"SEE PRODUCT"} />
      </BrowserRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});

it("Secondary Button matches snapshot", () => {
  const tree = renderer
    .create(
      <BrowserRouter>
        <SecondaryButton label={"SEE PRODUCT"} />
      </BrowserRouter>
    )
    .toJSON();

  expect(tree).toMatchSnapshot();
});
