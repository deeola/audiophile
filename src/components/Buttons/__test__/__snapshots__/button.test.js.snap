// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Primary Button matches snapshot 1`] = `
<a
  className="sc-bczRLJ lnGMsn"
  color="primary"
  hover="primary"
  href=""
  onClick={[Function]}
>
  SEE PRODUCT
</a>
`;

exports[`Secondary Button matches snapshot 1`] = `
<a
  className="sc-bczRLJ cwyJdT"
  color="secondary"
  hover="secondary"
  href=""
  onClick={[Function]}
>
  SEE PRODUCT
</a>
`;
